src/app/metadata-schema-form/metadata-form/metadata-form.component.spec.ts

```diff
@@ -1,104 +1,197 @@
+import {ComponentFixture, TestBed} from '@angular/core/testing';
 import {MetadataFormService} from '../metadata-form.service';
 import {JsonSchema} from '../models/json-schema';
 import {MetadataForm} from '../models/metadata-form';
 import {MetadataFormConfig} from '../models/metadata-form-config';
 import {MetadataFormTab} from '../models/metadata-form-layout';
 import {MetadataFormComponent} from './metadata-form.component';
-import SpyObj = jasmine.SpyObj;
-
-function getSchema() {
-  return {
-    'name': 'project'
-  } as JsonSchema;
-}
-
-function getData() {
-  return {
-    content: null,
-    admin: {
-      key: 'value'
-    }
-  };
-}
-
-function createTab(title) {
-  return {
-    title: title,
-    key: 'project.' + title,
-    items: [
-      'project.' + title
-    ]
-  } as MetadataFormTab;
-}
-
-function createMetadataFormConfig(tabs: MetadataFormTab[]) {
-  return {
-    viewMode: true,
-    removeEmptyFields: true,
-    layout: {
-      tabs: tabs
-    }
-  } as MetadataFormConfig;
-}
-
-function createFormGroup() {
-  return new FormGroup({
-    x: new FormControl('', Validators.required),
-    y: new FormControl('', Validators.required),
-    z: new FormControl('', Validators.required),
-  });
-}
-
-function createMetadataForm(tabs: MetadataFormTab[]): MetadataForm {
-  return {
-    key: 'test key',
-    jsonSchema: getSchema(),
-    config: createMetadataFormConfig(tabs),
-    metadataRegistry: jasmine.createSpyObj('MetadataRegistry', ['buildMetadataRegistry']),
-    formGroup: createFormGroup()
-  } as MetadataForm;
-}
+import {MetadataRegistry} from '@metadata-schema-form/models/metadata-registry';
 
 describe('MetadataFormComponent', () => {
   let component: MetadataFormComponent;
   let fixture: ComponentFixture<MetadataFormComponent>;
 
-  let metadataFormSvc: SpyObj<MetadataFormService>;
+  let metadataFormSvc;
+  let metadataRegistrySpy;
+  let formGroupSpy;
+
+  interface metadataFormDependencies {
+    key?: string;
+    schema?: JsonSchema;
+    config?: MetadataFormConfig;
+  }
 
-  const contentTab = createTab('content.contributors');
-  const adminTab = createTab('admin');
-  const tabs = [contentTab, adminTab];
+  function createTab(title) {
+    return {
+      title: title,
+      key: 'project.' + title,
+      items: [
+        'project.' + title
+      ]
+    } as MetadataFormTab;
+  }
 
-  beforeEach(async(() => {
-    metadataFormSvc = jasmine.createSpyObj('MetadataFormService', ['createForm']);
-    metadataFormSvc.createForm.and.returnValue(createMetadataForm(tabs));
+  function configureFormAndSvc({
+                                 key = 'test key',
+                                 schema = {'name': 'project'} as JsonSchema,
+                                 config = {}
+                               }: metadataFormDependencies) {
+    if (!config.layout) {
+      config.layout = {tabs: [createTab('tab')]};
+    }
+    if (!config.layout.tabs) {
```

Comment:
Is this not a duplicated version of the above lines?

Reply:
Good find, I'll fix that.